src/Transports.AspNetCore/GraphQLHttpMiddleware.cs

```diff
@@ -49,13 +48,13 @@ public class GraphQLHttpMiddleware<TSchema> where TSchema : ISchema
 
         private bool IsGraphQlRequest(HttpContext context)
         {
-            return context.Request.Path.StartsWithSegments(_options.Path);
+            return context.Request.Method.Equals("POST") && context.Request.Path.StartsWithSegments(_options.Path);
```

Comment:
Would this be better https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.httpmethods.ispost?view=aspnetcore-2.0 ?

Reply:
Good call!